lib/ash/resource/verifiers/verfiy_identities.ex

```diff
@@ -0,0 +1,22 @@
+defmodule Ash.Resource.Verifiers.VerifyIdentityFields do
+  @moduledoc """
+  Raises an error on potentially incompatible identity attributes.
+  """
+  use Spark.Dsl.Verifier
+
+  def verify(dsl) do
+    identities = Ash.Resource.Info.identities(dsl)
+
+    for identity <- identities do
+      for key <- identity.keys do
+        if !Ash.Resource.Info.attribute(dsl, key) do
+          raise Spark.Error.DslError,
+            module: Spark.Dsl.Verifier.get_persisted(dsl, :module),
+            message: "no such attribute #{inspect(key)}"
```

Comment:
lets add a `path` to the error, `path: [:identities, identity.name]`.

Additionally, let's make the error message a bit clearer: "All identity keys must be attributes. Got: #{inspect(key)}"

Reply:
done.